### ISSUE TITLE
Optional json schema object properties are required if they contain required properties

### ISSUE THREAD

**Issue by rasantel** [ngx-formly/ngx-formly#2752]
**Description**
If I specify a property that is not in the required list, and if that property is an object that contains a required property, Formly requires the required property of the optional object property to be entered.

The issue also affects non-required properties declared as `oneOf` and `anyOf` (I haven't tried `allOf`).

**Minimal Reproduction**

https://stackblitz.com/edit/angular-jb2v7c?file=src/assets/json-schema/optionalObjectRequiredProperty.json

I cloned this from the StackBlitz link on https://formly.dev/examples/advanced/json-schema and created new json schema examples to illustrate the issue.

The examples include a non-required object with a required property and also non-required properties with required fields within a `oneOf` selector.

**Your Environment**

- Angular version: 7.2.16
- Formly version: 5.10.15

**Additional context**
The following json schema results in a form that requires entering the `lorem` field for property `p` even though property `p` is not required:
```
{
    "type": "object",
    "properties": {
      "p": {
        "title": "Property P",
        "type": "object",
        "properties": {
          "lorem": {
            "title": "lorem",
            "type": "string"
          }
        },
        "required": [
          "lorem"
        ]
      }
    }
}
```
I tried two different json schema validators with this schema and an empty json, and the empty json is accepted: https://www.jsonschemavalidator.net/ and https://github.com/fge/json-schema-validator-demo .

Using selectors such as `oneOf` or `anyOf` has the same issue. For example:
```
{
    "type": "object",
    "properties": {
      "p": {
        "title": "Property P",
        "type": "object",
        "properties": {
          "lorem": {
            "title": "lorem",
            "type": "string"
          }
        },
        "required": [
          "lorem"
        ]
      }
    }
}
```

**Comment by EyPelz** [comment]
Also happens to me with allOf.

**Comment by aitboudad** [comment]
Our current approach is a little different here, whenever a field is required the parent object is considered required too, and from FORM perspective, I think it makes sense :thinking:.

Let think about the options we may have to solve this issue.

**Comment by tomer953** [comment]
Thanks for your reply,
I have the same problem as well.

I don't see why it makes sense from FORM perspective.
IMO it's totally acceptable to have required to child's elements of optional parent,
and the best proof is that the actual JSON Schema allows it,
meaning - not having this feature, cause us all the same bug. (the actual form is not the same as the json schema)

I can think of dozens examples of forms that have required properties if and only if you chose some checkbox or object.
ie you have to mention your child id, if and only if you chose to mention you have childs.

make sense?

**Comment by aitboudad** [comment]
> make sense?

Yes, just saying showing the required asterisk in the field while it's not required, can be confusing.

**Comment by tomer953** [comment]
I Agree, but in such a case, the parent field is not required, therefor should not have an asterisk, only the required child fields.

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v5.10.19](https://github.com/ngx-formly/ngx-formly/releases/tag/v5.10.19) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!

**Comment by rasantel** [comment]
thank you!
